src/ORM/Association/BelongsToMany.php

```diff
@@ -797,14 +797,19 @@ protected function _saveLinks(EntityInterface $sourceEntity, $targetEntities, $o
             $sourceKeys = array_combine($foreignKey, $sourceEntity->extract($bindingKey));
             $targetKeys = array_combine($assocForeignKey, $e->extract($targetPrimaryKey));
 
-            if ($sourceKeys !== $joint->extract($foreignKey)) {
-                $joint->set($sourceKeys, ['guard' => false]);
+            $changedKeys = (
+                $sourceKeys !== $joint->extract($foreignKey) ||
+                $targetKeys !== $joint->extract($assocForeignKey)
+            );
+            // Keys were changed, the junction table record _could_ be
```

Comment:
I wonder if we need so much explanation here, do we need to do something else or do it in another way, may be?

It _could_ be a sign of code smell or a general design problem...